src/Hooks/GutenbergCleanup.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace WPML\PB\Elementor\Hooks;
+
+use WPML\FP\Fns;
+use WPML\FP\Maybe;
+use WPML\FP\Obj;
+use WPML\FP\Relation;
+use WPML\PB\Elementor\DataConvert;
+use WPML_Elementor_Data_Settings;
+use function WPML\FP\partial;
+use function WPML\FP\pipe;
+
+class GutenbergCleanup implements \IWPML_Backend_Action, \IWPML_Frontend_Action, \IWPML_DIC_Action {
+
+	public function add_hooks() {
+		add_filter(
+			'update_post_metadata',
+			Fns::withoutRecursion( Fns::identity(), [ $this, 'removeGutenbergFootprint' ] ),
+			10,
+			4
+		);
+	}
+
+	/**
+	 * If we detect Gutenberg footprint in the Elementor data,
+	 * we'll remove it and delete the Gutenberg string package.
+	 *
+	 * @param null|bool $check
+	 * @param int       $postId
+	 * @param string    $metaKey
+	 * @param string    $metaValue
+	 *
+	 * @return mixed
+	 */
+	public function removeGutenbergFootprint( $check, $postId, $metaKey, $metaValue ) {
+		if (
+			WPML_Elementor_Data_Settings::META_KEY_DATA === $metaKey
+			&& WPML_Elementor_Data_Settings::is_edited_with_elementor( $postId )
+		) {
+			$data = DataConvert::unserialize( $metaValue );
+
+			// $ifDataHasChanged :: array -> bool
+			$ifDataHasChanged = function( $processedData ) use ( $data ) {
+				return json_encode( $processedData ) !== json_encode( $data );
+			};
```

Comment:
@strategio I would change this to make it more declarative where it's used.

```php
// $isDifferentFrom :: array -> array -> bool
$isDifferentFrom = curryN( 2, function( $data,  $processedData )  {
	return json_encode( $processedData ) !== json_encode( $data );
} );

...
...
->filter( $isDifferentFrom( $data ) )
```

Reply:
Done @brucepearson.